### ISSUE TITLE
Double clicking needed to add menu item

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#18329]
The flow right now seems to not need the extra step of setting me out to then click in to add again. Here is what happened:

![nav-throwout](https://user-images.githubusercontent.com/253067/68315966-8e4ee280-00b0-11ea-858b-a749caf7c753.gif)

**Comment by mtias** [comment]
This happens intermittently for me.

**Comment by shaunandrews** [comment]
I'm not able to reproduce this, and it seems more like a bug that a "design feedback" issue.

**Comment by mapk** [comment]
I'm able to reproduce this. However, I have no idea what's going on and am completely confused by this interaction.

![add-item 2019-11-06 16_42_14](https://user-images.githubusercontent.com/617986/68350213-7aa57b00-00b4-11ea-8b8b-103419682009.gif)

**Comment by getdave** [comment]
I have been able to replicate this and it's consistently one of the worst pieces of the interaction and UX. We'll look into the technical reasons behind this. Thanks for raising @karmatosed!

**Comment by draganescu** [comment]
I wonder if the quick fix isn't simply not showing the appender when a menu item is being edited?

**Comment by draganescu** [comment]
My initial idea was to hide the appender when the menu item is in edit mode. However the problem is that it is not immediately clear what to do to get the appender back.

So, I discovered the problem was the fact that when the appender is clicked the menu item UI shrinks because it is deselected and the appender moves away from below the cursor, and the focus event which should insert a new menu item doesn't fire anymore.

I made a PR that sets the appender absolutely positioned, but not sure this is the best approach yet. Any advice welcome.
